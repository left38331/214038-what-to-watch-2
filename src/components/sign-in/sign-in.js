import React from 'react';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';

import {Logo} from 'components/logo/logo';

const SignIn = (props) => {
  return props.isAuthorizationRequired ? <div className="user-page">
    <header className="page-header user-page__head">
      <Logo/>

      <h1 className="page-title user-page__title">Sign in</h1>
    </header>

    <div className="sign-in user-page__content">
      <form action="#" className="sign-in__form" onSubmit={props.formSubmitHandler}>
        <div className="sign-in__fields">
          <div className="sign-in__field">
            <input className="sign-in__input" type="email" placeholder="Email address" name="email" id="user-email" onChange={props.userInputHandler}/>
            <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
          </div>
          <div className="sign-in__field">
            <input className="sign-in__input" type="password" placeholder="Password" name="password" id="user-password" onChange={props.userInputHandler}/>
            <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
          </div>
        </div>
        <div className="sign-in__submit">
          <button className="sign-in__btn" type="submit">Sign in</button>
        </div>
      </form>
    </div>

    <footer className="page-footer">
      <Logo
        isFooter={true}
      />

      <div className="copyright">
        <p>Â© 2019 What to watch Ltd.</p>
      </div>
    </footer>
  </div> : <Redirect to="/"></Redirect>;
};

SignIn.propTypes = {
  formSubmitHandler: PropTypes.func.isRequired,
  userInputHandler: PropTypes.func.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
};

export {SignIn};
