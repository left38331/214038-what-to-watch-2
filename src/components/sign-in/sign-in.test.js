import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {SignIn} from 'components/sign-in/sign-in';

const props = {
  isAuthorizationRequired: false,
  formSubmitHandler: () => {},
  userInputHandler: () => {},
};

it(`Render correctly sign-in block component`, () => {
  const signIn = renderer.create(<BrowserRouter><SignIn {...props} /></BrowserRouter>).toJSON();

  expect(signIn).toMatchSnapshot();
});
