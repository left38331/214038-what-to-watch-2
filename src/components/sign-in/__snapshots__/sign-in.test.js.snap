// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly sign-in block component 1`] = `null`;
